### ISSUE TITLE
getting all transactions?

### ISSUE THREAD

**Issue by dmead** [plaid/plaid-node#145]
Hi

Is there an example of using the offset in getTransactions to get all the transactions for an auth token?

I may just not understand how to deal with this is promises, but it seems like something that isn't useful for the promise model when the only thing we have access to is the offset and the number of results back.

**Comment by ntindall** [comment]
@dmead some logic to this seems like it would work to me (note, I haven't tested it)

```typescript
const PAGE_SIZE = 100;

const getAllTransactions = async (): Array<Transaction> => {
  const $totalTransactions = [];

  let offset = 0;
  while (true) {
    const transactionsResponse = await getTransactions(access_token, start_date, end_date, {
      offset,
      count: PAGE_SIZE,
    });

    $totalTransactions.push(transactionsResponse.transactions);

    if ($totalTransactions.length == transactionsResponse.total_transactions) {
      break;
    }

    offset += transactionsResponse.transactions.length;
  }

  return $totalTransactions;
};
```

Perhaps it would be fruitful for us to provide a wrapper `getAllTransactions` that does this from the library. I imagine this is a common application.

**Comment by ntindall** [comment]
You can use the new helper method implemented in #148 to do this @dmead 

We will publish it with the next release.

**Comment by dmead** [comment]
@ntindall  thanks for this. it's pretty much what i ended up having to do. I will switch to using your function.
